neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/dialect/impl/BaseNeo4jEntityQueries.java

```diff
@@ -588,11 +588,11 @@ public String getUpdateEmbeddedColumnQuery(Object[] keyValues, String embeddedCo
 		return query;
 	}
 
-	public String getFindAssociationQuery(String role, AssociationKeyMetadata associationKeyMetadata) {
-		String query = findAssociationQueryCache.get( role );
+	public String getFindAssociationQuery(String relationshipType, AssociationKeyMetadata associationKeyMetadata) {
```

Comment:
@DavideD I'm going to merge the PR but it would be nice if you could add a comment to the method explaining the difference between the 2 and why we have to use one in one case and the other in the other case.